app/serializers/appointment_serializer.rb

```diff
@@ -1,6 +1,37 @@
 class AppointmentSerializer < ActiveModel::Serializer
   attributes :id, :start_time, :status
 
+  CANCELED = :CANCELED
+  PAST = :PAST
+  PRESENT = :PRESENT
+  PENDING = :PENDING
+
+  def status
+    if object.is_canceled
+      return CANCELED
+    end
+
+    curr_date = Time.now.to_date
+    appointment_date = object.start_time.to_date
+
+    if curr_date > appointment_date
+      PAST
+    elsif curr_date < appointment_date
+      return PENDING
+    else
+      curr_time = Time.now.to_time
+      start_time = object.start_time.to_time
+
+      if start_time > curr_time
+        return PENDING
+      elsif start_time < curr_time
+        return PAST
+      end
+
+      PRESENT
+    end
+  end
```

Comment:
I tried to refactor this a little bit

```suggestion
  def initialize(*args)
    super
    @curr_date = Time.now.to_date
    @curr_time = Time.now.to_time
  end

  def status
    return CANCELED if object.is_canceled?

    appointment_date = object.start_time.to_date
    start_time = object.start_time.to_time

    case
    when appointment_date < @curr_date
      PAST
    when appointment_date > @curr_date || start_time > @curr_time
      PENDING
    else
      PRESENT
    end
  end
```